templates/js-langchain/src/main.js

```diff
@@ -0,0 +1,86 @@
+import { Actor } from 'apify';
+import { ApifyDatasetLoader } from "langchain/document_loaders/web/apify_dataset";
+import { Document } from "langchain/document";
+import { HNSWLib } from "langchain/vectorstores/hnswlib";
+import { OpenAIEmbeddings } from "langchain/embeddings/openai";
+import { RetrievalQAChain } from "langchain/chains";
+import { OpenAI } from "langchain/llms/openai";
+
+import { retrieveVectorIndex, cacheVectorIndex } from './vector_index_cache.js';
+
+await Actor.init();
+
+const {
+    startUrls = [{ url: 'https://wikipedia.com' }],
+    maxCrawlPages = 3,
+    forceRecrawl = false, // Enforce a re-crawl of website content and re-creation of the vector index.
+    query = 'What is Wikipedia?',
+} = await Actor.getInput() || {};
+
+// There are 2 environment variables you need to configure (https://docs.apify.com/cli/docs/vars#set-up-environment-variables-in-apify-console):
+// 1. Apify API token you obtain at https://console.apify.com/account/integrations. This is not nessesary when running this Actor at Apify platform.
+// 2. OpenAI API key you obtain at https://platform.openai.com/account/api-keys.
+const { OPENAI_API_KEY, APIFY_API_TOKEN } = process.env;
+
+// Local directory where the vector index will be stored.
+const VECTOR_INDEX_PATH = './vector_index';
+
+if (!OPENAI_API_KEY || !OPENAI_API_KEY.length) throw new Error('Please configure the OPENAI_API_KEY environment variable!');
+if (!APIFY_API_TOKEN || !APIFY_API_TOKEN.length) throw new Error('Please configure the APIFY_API_TOKEN environment variable!');
```

Comment:
I would rather direct users to `apify login`, so that they don't have to set the env var manually. If you're logged in, the `APIFY_TOKEN` env var is set for you when you do `apify run`.